internal/expr/typeinfo.go

```diff
@@ -41,47 +41,52 @@ func typeInfo(value any) (*info, error) {
 }
 
 // generate produces and returns reflection information for the input
-// reflect.Value that is specifically required for SQLair operation.
-func generate(t reflect.Type) (*info, error) {
-	// Reflection information is only generated for structs.
-	if t.Kind() != reflect.Struct {
-		return nil, fmt.Errorf("internal error: attempted to obtain struct information for something that is not a struct: %s.", t)
-	}
-
-	info := info{
-		tagToField: make(map[string]field),
-		typ:        t,
-	}
-	tags := []string{}
-
-	for i := 0; i < t.NumField(); i++ {
-		f := t.Field(i)
-		// Fields without a "db" tag are outside of SQLair's remit.
-		tag := f.Tag.Get("db")
-		if tag == "" {
-			continue
+// reflect.Value that is specifically required for SQLAir operation.
+func generate(t reflect.Type) (info, error) {
```

Comment:
This also deserves a less generic name (generateTypeInfo, for example).

Reply:
Changed.